### DISCUSSION TITLE
Pipe coordinate system

### DISCUSSION THREAD

**Discussion by mostopalove** [antvis/G6#3630] [Q&A]
Please explain how exactly the pipe coordinates are calculated?
I need to align the elements inside the pipes and also move the pipes relative to each other.

**Comment by Yanyan-Wang** [comment]
Did you mean layout for nodes? Try the built-in layouts  https://g6.antv.vision/en/docs/api/graphLayout/guide

or custom a layout by registerLayout
https://g6.antv.vision/en/docs/api/registerLayout
https://g6.antv.vision/en/examples/net/layoutMechanism#customBigraph
